phoenix-core/src/main/java/org/apache/phoenix/util/CursorUtil.java

```diff
@@ -247,11 +269,11 @@ private void updateSelectValues(Tuple currentRow) throws SQLException {
                     PDataType type = projector.getExpression().getDataType();
                     Object value = projector.getValue(currentRow, type, new ImmutableBytesPtr());
                     if(value == null){
-                        whereExpressionNext = whereExpressionNext.replaceFirst(":"+Integer.toString(colBinding),"NULL");
-                        whereExpressionPrior = whereExpressionPrior.replaceFirst(":"+Integer.toString(colBinding),"NULL");
+                        whereExpressionNext = whereExpressionNext.replaceFirst("::"+Integer.toString(colBinding),"NULL");
+                        whereExpressionPrior = whereExpressionPrior.replaceFirst("::"+Integer.toString(colBinding),"NULL");
                     } else{
-                        whereExpressionNext = whereExpressionNext.replaceFirst(":"+Integer.toString(colBinding),formatString(value, type));
-                        whereExpressionPrior = whereExpressionPrior.replaceFirst(":"+Integer.toString(colBinding),formatString(value, type));
+                        whereExpressionNext = whereExpressionNext.replaceFirst("::"+Integer.toString(colBinding),formatString(value, type));
+                        whereExpressionPrior = whereExpressionPrior.replaceFirst("::"+Integer.toString(colBinding),formatString(value, type));
```

Comment:
I don't think , we need to modify queries for the this support.   You may need to remove this code.(And instead of replacing why we can't use prepared statements directly).